router/utils_test.go

```diff
@@ -9,68 +9,98 @@ import (
 
 	"github.com/AccumulateNetwork/accumulated/networks"
 
-	"github.com/AccumulateNetwork/accumulated/config"
+	cfg "github.com/AccumulateNetwork/accumulated/config"
 	"github.com/AccumulateNetwork/accumulated/internal/abci"
 	"github.com/AccumulateNetwork/accumulated/internal/chain"
 	"github.com/AccumulateNetwork/accumulated/internal/node"
 	"github.com/AccumulateNetwork/accumulated/types/state"
-	"github.com/spf13/viper"
+	tmcfg "github.com/tendermint/tendermint/config"
 	tmnet "github.com/tendermint/tendermint/libs/net"
 	"github.com/tendermint/tendermint/privval"
 )
 
-func randomRouterPorts() *config.Router {
+func randomRouterPorts() *cfg.Router {
 	port, err := tmnet.GetFreePort()
 	if err != nil {
 		panic(err)
 	}
-	return &config.Router{
+	return &cfg.Router{
 		JSONListenAddress: fmt.Sprintf("localhost:%d", port),
 		RESTListenAddress: fmt.Sprintf("localhost:%d", port+1),
 	}
 }
 
-func boostrapBVC(configfile string, workingdir string, baseport int) error {
-	idx := networks.IndexOf("Localhost")
-	net := networks.Networks[idx]
-	opts := node.InitOptions{}
+func initOptsForNetwork(t *testing.T, name string) node.InitOptions {
+	t.Helper()
+
+	// This will panic if `name` is wrong, causing the test to immediately fail
+	network := networks.Networks[networks.IndexOf(name)]
+	listenIP := make([]string, len(network.Nodes))
+	remoteIP := make([]string, len(network.Nodes))
+	config := make([]*cfg.Config, len(network.Nodes))
+
+	for i, net := range network.Nodes {
+		listenIP[i] = "tcp://0.0.0.0"
+		remoteIP[i] = net.IP
+		config[i] = new(cfg.Config)
+		config[i].Accumulate.Type = string(network.Type)
+
+		switch net.Type {
+		case cfg.Validator:
+			config[i].Config = *tmcfg.DefaultValidatorConfig()
+		case cfg.Follower:
+			config[i].Config = *tmcfg.DefaultValidatorConfig()
+		default:
+			fmt.Fprintf(os.Stderr, "Error: hard-coded network has invalid node type: %q\n", net.Type)
+			os.Exit(1)
+		}
+	}
+
+	return node.InitOptions{
+		ShardName: "accumulate.",
+		ChainID:   network.Name,
+		Port:      network.Port,
+		Config:    config,
+		RemoteIP:  remoteIP,
+		ListenIP:  listenIP,
+	}
+}
+
+func boostrapBVC(t *testing.T, configfile string, workingdir string) {
+	t.Helper()
+
+	opts := initOptsForNetwork(t, "Badlands")
 	opts.WorkDir = workingdir
-	opts.Port = net.Port
-	opts.ListenIP = make([]string, len(net.Nodes))
-	for i, node := range net.Nodes {
-		opts.ListenIP[i] = node.IP
```

Comment:
I called the `node.Init` argument `Options`, but none of it is optional. All properties must be populated for everything to work right. I added `initOptsForNetwork`, which does basically the same thing as `accumulated init`.